### PR TITLE
- added basic VFAT and NTFS support

### PR DESCRIPTION
**Pull request by aschnell**

### CODE REVIEW

File: storage/Devices/NtfsImpl.cc [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    Ntfs::Impl::do_set_label() const
+    {
+	const BlkDevice* blkdevice = get_blkdevice();
+
+	string cmd_line = NTFSLABELBIN " " + quote(blkdevice->get_name()) + " " + quote(get_label());
+	cout << cmd_line << endl;
+
+	SystemCmd cmd(cmd_line);
+	if (cmd.retcode() != 0)
+	    throw runtime_error("set-label ntfs failed");
```
Comment (shundhammer) [inline storage/Devices/NtfsImpl.cc]: use ST_THROW

File: storage/Devices/NtfsImpl.cc [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	const BlkDevice* blkdevice = get_blkdevice();
+
+	blkdevice->get_impl().wait_for_device();
+
+	string cmd_line = MKFSNTFSBIN " --fast --with-uuid " + quote(blkdevice->get_name());
+	cout << cmd_line << endl;
+
+	SystemCmd cmd(cmd_line);
+	if (cmd.retcode() != 0)
+	    throw runtime_error("create ntfs failed");
```
Comment (shundhammer) [inline storage/Devices/NtfsImpl.cc]: use ST_THROW

File: storage/Devices/VfatImpl.cc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	const BlkDevice* blkdevice = get_blkdevice();
+
+	blkdevice->get_impl().wait_for_device();
+
+	string cmd_line = MKFSFATBIN " " + quote(blkdevice->get_name());
+	cout << cmd_line << endl;
+
+	SystemCmd cmd(cmd_line);
+	if (cmd.retcode() != 0)
+	    throw runtime_error("create vfat failed");
```
Comment (shundhammer) [inline storage/Devices/VfatImpl.cc]: use ST_THROW

File: storage/Devices/VfatImpl.cc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    Vfat::Impl::do_set_label() const
+    {
+	const BlkDevice* blkdevice = get_blkdevice();
+
+	string cmd_line = FATLABELBIN " " + quote(blkdevice->get_name()) + " " + quote(get_label());
+	cout << cmd_line << endl;
+
+	SystemCmd cmd(cmd_line);
+	if (cmd.retcode() != 0)
+	    throw runtime_error("set-label vfat failed");
```
Comment (shundhammer) [inline storage/Devices/VfatImpl.cc]: use ST_THROW

**Comment by shundhammer** [comment]
Other than the throw/ST_THROW issues, LGTM.

**Comment by aschnell** [comment]
OK, using new ST_THROW macro (also at a few other places).

**Pull request merged**